### PR TITLE
fix move toppings controls and ui edit

### PR DESCRIPTION
**Pull request by jermsam**

### CODE REVIEW

File: src/Editor.tsx [hunk 1]
```diff
@@ -1,57 +1,82 @@
...
+
+                                changeDoc(
+                                    (doc) => {
+                                        const replacedLayer = doc.layers[key - 1];
+                                        const untouched = doc.layers.slice(key + 1)
+                                        const frontLayer = [...doc.layers]
+                                        frontLayer.splice(key - 1, untouched.length + 1, layer)
+                                        frontLayer.push(...[...untouched])
+                                        frontLayer[key] = replacedLayer
+                                        doc.layers = frontLayer
```
Comment (neftaly) [inline src/Editor.tsx]: You have to mutate the array in-place for automerge, immutable methods ([...array]) don't provide it with the change information it needs

File: src/Burger.tsx [hunk 2]
```diff
@@ -37,6 +38,6 @@ export const Layer = ({ type }) => {
 // As we're building from the bottom up, reverse the layer order
 export const Burger = ({ layers = [] }) => (
   <Flex justifyContent="center" alignItems="center" position={[-0.5, 0, 0]}>
-    {layers.map((type, key) => <Layer key={key} type={type} />).reverse()}
+    {layers.map((type, key) => <Layer key={key} type={type} />)}
```
Comment (neftaly) [inline src/Burger.tsx:41]: Are you reversing the array in the editor instead?

**Review by neftaly — CHANGES_REQUESTED** [review]

File: src/Editor.tsx (hunk 1 above)
Comment (jermsam) [reply src/Editor.tsx]: I mutate as the last step

File: src/Burger.tsx (hunk 2 above)
Comment (jermsam) [reply src/Burger.tsx:41]: yeah

File: src/Editor.tsx (hunk 1 above)
Comment (neftaly) [reply src/Editor.tsx]: ok, I don't have the mental capacity to do this at the moment sorry :(

Comment (jermsam) [reply src/Editor.tsx]: it's ok, you can look into it when you can

Comment (jermsam) [reply src/Editor.tsx]: but I've tested it and it works fine ... see video

Comment (jermsam) [reply src/Editor.tsx]: https://github.com/neftaly/autoburger/assets/11094475/b383fde8-5583-4c4e-94f1-7b62a564b508

File: src/Editor.tsx [hunk 3]
```diff
@@ -1,57 +1,70 @@
...
+    return (
+        <div>
+            <hr/>
+            <div style={{padding: '1rem'}}>
+                {Object.entries(toppings).map(([id]) => (
+                    <button
+                        key={id}
+                        style={{display: "block", marginBottom: '0.3rem'}}
+                        children={`+ ${toppings[id].name}`}
+                        onClick={() => changeDoc((doc) => void doc.layers.unshift(id))}
```
Comment (neftaly) [inline src/Editor.tsx:13]: oic this is instead of .reverse, good idea

**Review by neftaly — APPROVED** [review]

**Pull request merged**